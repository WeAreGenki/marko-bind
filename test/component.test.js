'use strict'; // eslint-disable-line

const component = require('./fixtures/component');

describe('component', () => {
  it('should have a renderToString function', () => {
    expect(component.renderToString).toBeDefined();
  });

  it('should render correctly', () => {
    component.renderToString({
      id: 'txtMoo',
      label: 'Moo',
      placeholder: 'E.g: cow',
    }, (err, html) => {
      if (err) throw err;
      expect(html).toMatchSnapshot();
    });
  });

  // eslint-disable-next-line jest/no-disabled-tests
  it.skip('should have an input', async () => {
    const renderResult = await component.render({
      id: 'oom',
      label: 'aaa',
      placeholder: 'aaa',
    });
    const browserContext = renderResult.appendTo(document.body).getComponent();
    expect(browserContext.input).toBeDefined();
  });

  // eslint-disable-next-line jest/no-disabled-tests
  it.skip('should getValue() correctly', async () => {
    const renderResult = await component.render({
      id: 'woo',
      label: 'bbb',
      placeholder: 'bbb',
    });
    const browserContext = renderResult.appendTo(document.body).getComponent();
    expect(browserContext.getValue()).toBe('woo');
  });
});
