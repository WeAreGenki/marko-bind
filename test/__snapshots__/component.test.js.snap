// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should render correctly 1`] = `
<!--M#s0-->
<div class="field">
  <label class="field_label" for="s0-txtMoo">Moo</label><span class="field_control field_control--textbox"><input type="text" placeholder="E.g: cow" class="textbox" id="s0-txtMoo"></span></div>
<!--M/s0-->
`;
